### ISSUE TITLE
why is the test data used for creating the quantiles?

### ISSUE THREAD

**Issue by aanilpala** [tensorflow/lattice#21]
https://github.com/tensorflow/lattice/blob/master/examples/uci_census.py#L291

I don't get why the test data is used for creating the quantiles which is supposed to take place only during the training. Typically, you shouldn't even take a look at the test set during the training.

Now on a second thought, I am guessing it might have to do with making sure that the when testing the input features will always be within the the quantiles.

**Comment by janpfeifer** [comment]
Good catch! This is definitely it should read from the train dataset (not to cheat on the test dataset, as you pointed out).

And about points outside the seen range, that should be fine. The library takes care of it, either by extrapolating and/or clipping. (* I need to verify, I think right now it only clips)

Plus one should add documentation to [--create_quantiles](https://github.com/tensorflow/lattice/blob/master/examples/uci_census.py#L107) flag explicitly stating where the data will be read from (if from --test or the --train [datasets](https://github.com/tensorflow/lattice/blob/master/examples/uci_census.py#L85
))
